src/libstd/sys/redox/fs.rs

```diff
@@ -447,7 +447,10 @@ pub fn stat(p: &Path) -> io::Result<FileAttr> {
 }
 
 pub fn lstat(p: &Path) -> io::Result<FileAttr> {
-    stat(p)
+    let fd = cvt(syscall::open(p.to_str().unwrap(),
+                               syscall::O_CLOEXEC | syscall::O_STAT | syscall::O_NOFOLLOW))?;
```

Comment:
Ah I see `O_STAT` is not posix, I guess that justifies my 'ish' in 'posixish'. Thanks for the clarification.